chemprop/featurizers/molecule.py

```diff
@@ -41,3 +49,62 @@ class MorganBinaryFeaturizer(MorganFeaturizerMixin, BinaryFeaturizerMixin, Vecto
 @MoleculeFeaturizerRegistry("morgan_count")
 class MorganCountFeaturizer(MorganFeaturizerMixin, CountFeaturizerMixin, VectorFeaturizer[Mol]):
     pass
+
+
+@MoleculeFeaturizerRegistry("rdkit_2d")
+class RDKit2DFeaturizer(VectorFeaturizer[Mol]):
+    def __init__(self):
+        warnings.warn(
+            "The RDKit 2D features can deviate signifcantly from a normal distribution. Consider "
+            "manually scaling them using an appropriate scaler before creating datapoints, rather "
+            "than using the scikit-learn `StandardScaler` (the default in Chemprop)."
+        )
+
+    def __len__(self) -> int:
+        return len(Descriptors.descList)
+
+    def __call__(self, mol: Chem.Mol) -> np.ndarray:
+        features = np.array(
+            [
+                func(mol)
+                for _, func in filter(
+                    lambda i: i[0] != "SPS" or mol.GetNumHeavyAtoms() > 0, Descriptors.descList
```

Comment:
```suggestion
            0.0 if name == "SPS" and mol.GetNumHeavyAtoms() == 0 else func(mol)
            for name, func in Descriptors.descList
```
For species without heavy atoms, the length of RDKit2D will be shorter by 1. I think it should be padded with 0.

Reply:
Thanks for catching that!